### PR TITLE
User Management Navigator Supervisor Backend Implementation

### PR DESCRIPTION
**Pull request by sanjaytkbabu**
# Description
[PADS-222](https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-222)
User Management - Supervisor backend implementation
features - create new user, revoke user, manage user (will be functional after rbac)
migration added for access_request table

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Subin1Doo** [comment]
Things are looking great! Few comments.. 
For the status can we say "what" it's pending. So it would either say "Pending approval" or "Pending revocation."
And for the Create new user modal, can we make sure everything is left / right aligned. Thanks!

**Comment by sanjaytkbabu** [comment]
> Things are looking great! Few comments.. For the status can we say "what" it's pending. So it would either say "Pending approval" or "Pending revocation." And for the Create new user modal, can we make sure everything is left / right aligned. Thanks!

done!

**Comment by Subin1Doo** [comment]
Sorry just double checking, once the pending revocation has been approved by the admin the status should say "revoked" in the Nav supervisor view. "Approved" for new user approval.

**Comment by sanjaytkbabu** [comment]
> Sorry just double checking, once the pending revocation has been approved by the admin the status should say "revoked" in the Nav supervisor view. "Approved" for new user approval.

yes, that's correct

File: app/src/utils/enums/application.ts
Comment (kyle1morel) [inline app/src/utils/enums/application.ts:33]: Misnomer. This is the status of the access request, not the user.

File: app/src/utils/enums/application.ts
Comment (kyle1morel) [inline app/src/utils/enums/application.ts:38]: Style: Please ensure new lines are between blocks. I've noticed this a few times within this PR.

File: app/src/db/models/access_request.ts
Comment (sanjaytkbabu) [reply app/src/db/models/access_request.ts]: thanks for catching

File: app/src/services/user.ts
Comment (sanjaytkbabu) [reply app/src/services/user.ts]: done

File: app/src/services/user.ts
Comment (sanjaytkbabu) [reply app/src/services/user.ts]: thanks for catching

File: app/src/services/user.ts
Comment (sanjaytkbabu) [reply app/src/services/user.ts]: I just put it there to stop unused 'user' variable complaints. I've removed it!

File: app/src/utils/enums/application.ts
Comment (sanjaytkbabu) [reply app/src/utils/enums/application.ts]: makes sense

File: app/src/utils/enums/application.ts
Comment (sanjaytkbabu) [reply app/src/utils/enums/application.ts]: done

**Comment by Subin1Doo** [comment]
Comment on the sorting of the "status" column. 
When I sort this column, it doesn't really seem to be sorting. The order seems rather random. 
Can we make it so that when they sort by "status" it shows the pending approval ones first, then the pending revocation ones, then all the approved ones? and the other way around. Thank you!

File: app/src/services/user.ts
Comment (sanjaytkbabu) [reply app/src/services/user.ts:141]: done

**Comment by sanjaytkbabu** [comment]
> Comment on the sorting of the "status" column. When I sort this column, it doesn't really seem to be sorting. The order seems rather random. Can we make it so that when they sort by "status" it shows the pending approval ones first, then the pending revocation ones, then all the approved ones? and the other way around. Thank you!

As discussed went with PR, PA & A.

**Review by kyle1morel — APPROVED** [review]
Reviewed semantically. Lots of file changes, difficult to follow the full logic. Functionality looks to be there.

**Pull request merged**
